easybuild/framework/easystack.py

```diff
@@ -0,0 +1,218 @@
+from easybuild.tools.filetools import read_file
+try:
+    import yaml
+except ImportError:
+    pass
+_log = fancylogger.getLogger('easystack', fname=False)
+
+
+class EasyStack(object):
+    """One class instance per easystack. General options + list of all SoftwareSpecs instances"""
+    def __init__(self):
+        self.easybuild_version = None
+        self.robot = False
+        self.software_list = []
+
+    def compose_ec_filenames(self):
+        """Returns a list of all easyconfig names"""
+        ec_filenames = []
+        for sw in self.software_list:
+            full_ec_version = det_full_ec_version({
+                'toolchain': {'name': sw.toolchain_name, 'version': sw.toolchain_version},
+                'version': sw.version,
+                'versionsuffix': sw.versionsuffix,
+            })
+            ec_filename = '%s-%s.eb' % (sw.name, full_ec_version)
+            ec_filenames.append(ec_filename)
+        return ec_filenames
+
+    # flags applicable to all sw (i.e. robot)
+    def get_general_options(self):
+        """Returns general options (flags applicable to all sw (i.e. --robot))"""
+        general_options = {}
+        # TODO add support for general_options
+        # general_options['robot'] = self.robot
+        # general_options['easybuild_version'] = self.easybuild_version
+        return general_options
+
+
+class SoftwareSpecs(object):
+    """Contains information about every software that should be installed"""
+    def __init__(self, name, version, versionsuffix, toolchain_version, toolchain_name):
+        self.name = name
+        self.version = version
+        self.toolchain_version = toolchain_version
+        self.toolchain_name = toolchain_name
+        self.versionsuffix = versionsuffix
+
+
+class EasyStackParser(object):
+    """Parser for easystack files (in YAML syntax)."""
+    @staticmethod
+    def parse(filepath):
+        """Parses YAML file and assigns obtained values to SW config instances as well as general config instance"""
+        f = read_file(filepath)
+        easystack_raw = yaml.safe_load(f)
+        easystack = EasyStack()
+
+        try:
+            software = easystack_raw["software"]
+        except (KeyError):
+            wrong_structure_file = "Not a valid EasyStack YAML file: no 'software' key found"
+            raise EasyBuildError(wrong_structure_file)
+
+        # assign software-specific easystack attributes
+        for name in software:
+            name = str(name)
+            try:
+                toolchains = software[name]['toolchains']
+            except (KeyError):
+                raise EasyBuildError("Toolchains for software '%s' are not defined" % name)
+            for toolchain in toolchains:
+                toolchain = str(toolchain)
+                try:
+                    toolchain_name = toolchain.split('-', 1)[0]
+                    toolchain_version = toolchain.split('-', 1)[1]
```

Comment:
Only split once, ensure you get the correct number of parts, complain if not (and no need for `try`):

```python
toolchain_parts = toolchain.split('-', 1)
if len(toolchain_parts) == 2:
    toolchain_name, toolchain_version = toolchain_parts
elif len(toolchain_parts) == 1:
    toolchain_name, toolchain_version = toolchain, ''
else:
    raise EasyBuildError("Incorrect toolchain specification, too many parts: %s", toolchain_parts)
```